script/integration.go

```diff
@@ -50,17 +50,31 @@ func mainIntegration() {
 	for _, file := range files {
 		tests <- file
 	}
+	close(tests)
+
+	outputDone := make(chan bool)
+	go func() {
```

Comment:
Nit: this could be replaced with:

```go
go func() {
        for out := range output {
                fmt.Println(out)
        }
        outputDone <- true
}()
```

Though I am curious why this change is included in this PR?

Reply:
I was running into an issue where `script/integration` wouldn't dump out the `test-pre-push.sh` errors for this PR. This keeps the process alive until the tests have run AND the output has been sent STDOUT.

Reply:
I had planned on pushing that out as a separate PR. I must've accidentally merged to my local master, so it was still there when I branched off for this PR :)

I'll port the `Refspec()` change to #2734, and push out a PR for the test change, and another for this change.

Reply:
Rad! Thanks for porting this into #2734, I think that should be ready to go now 👍 .